app/Services/NetworkDataService.php

```diff
@@ -48,17 +48,26 @@ private function processClients(array $clients): void
 
     /**
      * If any aircraft has passed the timeout window, remove it from the list.
+     *
+     * NOTE: Events should always fire before final deletion because the listeners
```

Comment:
Could you hook into the model deleting event for this?

Reply:
It would be the net same effect (event fires immediately prior to deletion) - however we'd have to define all the events as `NetworkAircraft::deleting()` on every app invocation, which arguably puts the events too close to the models, or we'd have to listen for every single eloquent deleting event which is also inefficient

Reply:
Yep seems sound to me - cheers for that